backend/firebaseFunctions.js

```diff
@@ -70,28 +70,76 @@ export async function createCircle(data) {
 	data.admin[`${UID}`] = true;
 	data.owner = UID;
 
+	let circles = Object.keys((await readData(`prayer_circle/circles`)) || {});
+	const allAdminCodes = new Array(circles.length);
+	const allPublicCodes = new Array(circles.length);
+	for (let i = 0; i < circles.length; i++) {
+		let circle = circles[i];
+		let circleData =
+			(await readData(`prayer_circle/circles/${circle}`)) || {};
+		let circleStruct = {
+			adminCode: circleData.adminCode,
+			publicCode: circleData.publicCode
+		};
+		allAdminCodes[i] = circleStruct.adminCode;
+		allPublicCodes[i] = circleStruct.publicCode;
+	}
+
+	for (let i = 0; i < 2; i++) {
+		let unusedCode = false;
+		while (!unusedCode) {
+			unusedCode = true;
+			if (i === 0) {
+				data.adminCode = Math.floor(
+					Math.random() * 90000000 + 10000000
+				);
+				for (let j = 0; j < allAdminCodes.length; j++) {
+					if (allAdminCodes[j] === data.adminCode) {
+						unusedCode = false;
+					}
+				}
+			} else {
+				data.publicCode = Math.floor(
+					Math.random() * 90000000 + 10000000
+				);
+				if (data.publicCode === data.adminCode) {
+					unusedCode = false;
+				} else {
+					for (let j = 0; j < allPublicCodes.length; j++) {
+						if (allPublicCodes[j] === data.publicCode) {
+							unusedCode = false;
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
This part of the code can also be simplified a lot. The first thing we can do is have one loop. Outside of the loop, we should set up the codes like this:
`adminCode = 0;
publicCode = 0;`
then we can loop until we find a unique value for both, similar to what you have already. that would look like:
`while (adminCode == 0 && publicCode == 0) {
    if(adminCode == 0) {
        let tempAdminCode = Math.random() * 90000000 + 10000000;
        if(!codes.includes(tempAdminCode)) {
            adminCode = tempAdminCode;
            codes.push(tempAdminCode);
        }
    }
    if(publicCode == 0) {
        let tempPublicCode = Math.random() * 90000000 + 10000000;
        if(!codes.includes(tempPublicCode)) {
            publicCode = tempPublicCode;
            codes.push(tempPublicCode);
        }
    }
}`

Reply:
look like that didn't format right, but copy pasting into vscode should make it readable